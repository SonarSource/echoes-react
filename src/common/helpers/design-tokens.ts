/*
 * Echoes React
 * Copyright (C) 2023-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import designTokensColors from '~generated/design-tokens-colors.json';

export type DesignTokens = keyof typeof designTokensColors;

export type DesignTokensColors = {
  [K in DesignTokens]: K extends `echoes-color-${string}` ? K : never;
}[DesignTokens];
