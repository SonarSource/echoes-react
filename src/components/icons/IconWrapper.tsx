/*
 * Echoes React
 * Copyright (C) 2023-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import styled from '@emotion/styled';

export type IconProps = Omit<JSX.IntrinsicElements['span'], 'children'>;

export function IconWrapper(props: Readonly<IconProps & { children: React.ReactNode }>) {
  const { children, className, ...rest } = props;

  return (
    <StyledSpan className={className} {...rest}>
      {children}
    </StyledSpan>
  );
}

const StyledSpan = styled.span`
  font-family: 'Material Symbols Rounded';
  font-size: 1.2em;
  font-weight: normal;
  font-style: normal;
  line-height: 1;
  letter-spacing: normal;
  text-transform: none;
  white-space: nowrap;
  word-wrap: normal;
  direction: ltr;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-rendering: optimizeLegibility;
`;
