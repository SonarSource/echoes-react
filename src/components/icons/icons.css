@font-face {
  font-family: 'Material Symbols Rounded';
  font-style: normal;
  font-weight: 100 700;
  font-display: block;
  src: url('material-symbols/material-symbols-rounded.woff2') format('woff2');
}

@font-face {
  font-family: 'Echoes Icons';
  font-style: normal;
  font-weight: 400;
  font-display: auto;
  src: url('./echoes-icons.woff2') format('woff2');
}
