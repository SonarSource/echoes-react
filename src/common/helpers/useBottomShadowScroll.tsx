/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import styled from '@emotion/styled';
import { RefObject, useCallback, useEffect, useState } from 'react';
import { cssVar } from '~utils/design-tokens';
import { useResizeObserver } from './useResizeObserver';

/**
 * Custom hook to manage the visibility of a bottom shadow on a scrollable container.
 * The shadow indicates that there is more content to scroll to.
 * @param scrollableContainerRef - Ref to the scrollable container element, its height and scroll event are monitored to recompute the shadow
 * @param resizableContentRef - Ref to the resizable content element inside the scrollable container, its height is monitored to recompute the shadow
 * @returns {boolean} Whether the bottom shadow should be shown
 */
export function useBottomShadowScroll(
  scrollableContainerRef: RefObject<HTMLElement>,
  resizableContentRef: RefObject<HTMLElement>,
) {
  const [showBottomShadow, setShowBottomShadow] = useState(false);

  // Monitor the height of the scrollable container, and recompute the shadow if it changes
  const { height: scrollableContainerHeight } = useResizeObserver(scrollableContainerRef);
  // Monitor the height of the resizable content, and recompute the shadow if it changes, because it may move the scroll position
  const { height: resizableContentHeight } = useResizeObserver(resizableContentRef);

  const updateBottomShadowScroll = useCallback(() => {
    if (scrollableContainerRef.current) {
      const { scrollHeight, clientHeight, scrollTop } = scrollableContainerRef.current;
      setShowBottomShadow(scrollTop + clientHeight < scrollHeight - 8); // -8px for bottom padding
    }
  }, [scrollableContainerRef]);

  // Compute the bottom shadow visibility on first render and attach the scroll event listener
  useEffect(() => {
    const scrollableContainerRefNode = scrollableContainerRef.current;
    if (!scrollableContainerRefNode) {
      return undefined;
    }

    // Re-compute the bottom shadow visibility on scroll event
    scrollableContainerRefNode.addEventListener('scroll', updateBottomShadowScroll);

    // Initial computation
    updateBottomShadowScroll();

    return () => {
      scrollableContainerRefNode.removeEventListener('scroll', updateBottomShadowScroll);
    };
  }, [scrollableContainerRef, updateBottomShadowScroll]);

  // Re-compute the bottom shadow visibility on container and content resize
  useEffect(updateBottomShadowScroll, [
    resizableContentHeight,
    scrollableContainerHeight,
    updateBottomShadowScroll,
  ]);

  return [showBottomShadow] as const;
}

export const BottomShadowScroll = styled.div`
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;

  min-height: ${cssVar('dimension-height-400')};

  border-bottom: ${cssVar('border-width-default')} solid ${cssVar('color-border-weak')};
  background: radial-gradient(
      farthest-side at 50% 100%,
      ${cssVar('bottom-scroll-colors-shadow-gradient')},
      ${cssVar('color-support-transparent')}
    )
    center bottom;
`;
BottomShadowScroll.displayName = 'BottomShadowScroll';
