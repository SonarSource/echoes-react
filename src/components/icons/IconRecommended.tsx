/*
 * Echoes React
 * Copyright (C) 2023-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { forwardRef } from 'react';
import { IconFilledProps, IconMaterialWrapper } from './IconWrapper';

export const IconRecommended = forwardRef<HTMLSpanElement, IconFilledProps>((props, ref) => {
  // This is Material Symbols' "new_releases" icon
  return (
    <IconMaterialWrapper {...props} ref={ref}>
      &#xE031;
    </IconMaterialWrapper>
  );
});
IconRecommended.displayName = 'IconRecommended';
