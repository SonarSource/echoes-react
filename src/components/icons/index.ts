/*
 * Echoes React
 * Copyright (C) 2023-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import './icons.css';

export type { IconProps } from './IconWrapper';

export { IconArrowDownRight } from './IconArrowDownRight';
export { IconArrowLeft } from './IconArrowLeft';
export { IconArrowRight } from './IconArrowRight';
export { IconArrowUpRight } from './IconArrowUpRight';
export { IconBell } from './IconBell';
export { IconBranch } from './IconBranch';
export { IconBug } from './IconBug';
export { IconCalendar } from './IconCalendar';
export { IconCheck } from './IconCheck';
export { IconCheckCircle } from './IconCheckCircle';
export { IconChevronDown } from './IconChevronDown';
export { IconClock } from './IconClock';
export { IconCodeSmell } from './IconCodeSmell';
export { IconCollapse } from './IconCollapse';
export { IconComment } from './IconComment';
export { IconCopy } from './IconCopy';
export { IconDash } from './IconDash';
export { IconDirectory } from './IconDirectory';
export { IconDot } from './IconDot';
export { IconDownload } from './IconDownload';
export { IconEqual } from './IconEqual';
export { IconError } from './IconError';
export { IconExpand } from './IconExpand';
export { IconFile } from './IconFile';
export { IconFileCode } from './IconFileCode';
export { IconFilter } from './IconFilter';
export { IconGear } from './IconGear';
export { IconGitBranch } from './IconGitBranch';
export { IconHome } from './IconHome';
export { IconInfo } from './IconInfo';
export { IconInheritance } from './IconInheritance';
export { IconLink } from './IconLink';
export { IconLinkExternal } from './IconLinkExternal';
export { IconLock } from './IconLock';
export { IconMegaphone } from './IconMegaphone';
export { IconMoreVertical } from './IconMoreVertical';
export { IconOverview } from './IconOverview';
export { IconPeople } from './IconPeople';
export { IconPin } from './IconPin';
export { IconPlus } from './IconPlus';
export { IconProject } from './IconProject';
export { IconPullrequest } from './IconPullRequest';
export { IconQuestionMark } from './IconQuestionMark';
export { IconRecommended } from './IconRecommended';
export { IconRocket } from './IconRocket';
export { IconSearch } from './IconSearch';
export { IconSecurityFinding } from './IconSecurityFinding';
export { IconSlideshow } from './IconSlideshow';
export { IconStar } from './IconStar';
export { IconStatusConfirmed } from './IconStatusConfirmed';
export { IconStatusOpen } from './IconStatusOpen';
export { IconStatusReopened } from './IconStatusReopened';
export { IconStatusResolved } from './IconStatusResolved';
export { IconTarget } from './IconTarget';
export { IconUnfold } from './IconUnfold';
export { IconUnfoldDown } from './IconUnfoldDown';
export { IconUnfoldUp } from './IconUnfoldUp';
export { IconVulnerability } from './IconVulnerability';
export { IconWarning } from './IconWarning';
export { IconX } from './IconX';
