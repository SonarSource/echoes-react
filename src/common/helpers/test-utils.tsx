/*
 * Echoes React
 * Copyright (C) 2023-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { RenderOptions, render as rtlRender } from '@testing-library/react';
import userEvent, { Options as UserEventsOptions } from '@testing-library/user-event';
import React, { ComponentProps, PropsWithChildren } from 'react';
import { IntlProvider } from 'react-intl';
import { PropsWithLabels } from '~types/utils';
import { TooltipProvider } from '../../components/tooltip';

export function render(
  ui: React.ReactElement,
  options?: RenderOptions,
  userEventOptions?: UserEventsOptions,
) {
  return {
    ...rtlRender(ui, { wrapper: ContextWrapper, ...options }),
    user: userEvent.setup(userEventOptions),
  };
}

function ContextWrapper({ children }: PropsWithChildren<{}>) {
  return (
    <IntlProvider defaultLocale="en-us" locale="en-us">
      <TooltipProvider delayDuration={0}>{children}</TooltipProvider>
    </IntlProvider>
  );
}

export type OmitPropsWithLabels<T extends React.JSXElementConstructor<any>> = Partial<
  Omit<ComponentProps<T>, keyof PropsWithLabels<{}>>
> &
  PropsWithLabels<{}>;
