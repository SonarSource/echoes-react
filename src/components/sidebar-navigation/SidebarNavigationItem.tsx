/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import classNames from 'classnames';
import { forwardRef, ForwardRefExoticComponent, MouseEventHandler } from 'react';
import { NavLinkBase, NavLinkBaseProps } from '~common/components/NavLinkBase';
import { TextNode } from '~types/utils';
import { cssVar } from '~utils/design-tokens';
import { IconProps } from '../icons';
import { Tooltip } from '../tooltip';
import {
  itemIconStyles,
  sidebarNavigationBaseItemStyles,
  SidebarNavigationItemLabel,
} from './SidebarNavigationItemStyles';

export interface SidebarNavigationItemProps
  extends Pick<NavLinkBaseProps, 'isMatchingFullPath' | 'enableOpenInNewTab' | 'to'> {
  ariaLabel?: string;
  /**
   * The label of the SidebarNavigationItem.
   * It can be a string or a JSX.Element, in the case of a JSX.Element it should not be wrapped in
   * a `<Text>` component, the SidebarNavigationItem already handles the typography styling for you.
   */
  children: TextNode;
  className?: string;
  /**
   * Whether to display the tooltip on the item or not.
   * By default the tooltip is disabled, it should only be enabled if you expect the content to be ellipsed.
   */
  enableTooltip?: boolean;
  /**
   * Control whether the SidebarNavigationItem is active or not.
   * If true, the item will have a different style to indicate it is active.
   *
   * By default this behavior is handled by the underlying react-router's NavLink component,
   * overriding this is only needed for complex scenarios.
   */
  isActive?: boolean;
  /**
   * The onClick handler for the SidebarNavigationItem.
   */
  onClick?: MouseEventHandler<HTMLAnchorElement>;
  /**
   * The icon component to display at the start of the SidebarNavigationItem.
   * Must be an Echoes Icon component.
   */
  Icon?: ForwardRefExoticComponent<IconProps & React.RefAttributes<HTMLSpanElement>>;
}

export const SidebarNavigationItem = forwardRef<HTMLAnchorElement, SidebarNavigationItemProps>(
  (props, ref) => {
    const { children, className, enableTooltip, Icon, isActive = false, ...htmlProps } = props;

    return (
      <Tooltip content={enableTooltip ? children : undefined} side="right">
        <NavigationItem
          {...htmlProps}
          className={classNames({ active: isActive }, className)}
          ref={ref}>
          {Icon ? <Icon css={navigationItemIconStyles} /> : undefined}

          <SidebarNavigationItemLabel>{children}</SidebarNavigationItemLabel>
        </NavigationItem>
      </Tooltip>
    );
  },
);

SidebarNavigationItem.displayName = 'SidebarNavigationItem';

const NavigationItem = styled(NavLinkBase)`
  ${sidebarNavigationBaseItemStyles}

  // When the item is inside an accordion, the display value change based on the state of the accordion, hidding the items when collapsed
  // This css property is set by the SidebarNavigationAccordionItem component
  // Fallback to flex if not inside an accordion
  display: var(--sidebar-navigation-accordion-children-display, flex);

  &:active,
  &.active {
    // Always display the item when active even if behind a collapsed accordion, this override the previously set display value from the css property
    display: flex;

    background-color: ${cssVar('color-background-selected-weak-default')};
    color: ${cssVar('color-text-accent')};
    font: ${cssVar('typography-text-default-semi-bold')};

    &:hover {
      background-color: ${cssVar('color-background-selected-weak-hover')};
    }
  }
`;
NavigationItem.displayName = 'NavigationItem';

const navigationItemIconStyles = css`
  ${itemIconStyles}

  ${NavigationItem}.active > &,
  ${NavigationItem}:active > & {
    color: ${cssVar('color-icon-accent')};
  }
`;
