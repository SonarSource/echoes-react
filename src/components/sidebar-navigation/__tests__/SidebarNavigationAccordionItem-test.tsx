/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { matchers } from '@emotion/jest';
import { screen } from '@testing-library/react';
import { renderWithMemoryRouter } from '~common/helpers/test-utils';
import { IconBranch } from '../../icons';
import {
  SidebarNavigationAccordionItem,
  SidebarNavigationAccordionItemProps,
} from '../SidebarNavigationAccordionItem';
import { SidebarNavigationItem } from '../SidebarNavigationItem';

expect.extend(matchers);

// Mock the useIsOverflow hook
jest.mock('~common/helpers/useIsOverflow', () => ({
  useIsOverflow: jest.fn().mockReturnValue([false]),
}));

it('should expand hidden elements when clicked', async () => {
  const onOpen = jest.fn();
  const onClose = jest.fn();
  const { user } = setupSidebarNavigationAccordionItem({ onOpen, onClose });

  const accordionButton = screen.getByRole('button', { name: 'Accordion Item' });
  expect(accordionButton).toBeInTheDocument();
  checkAccordionPanelVisibility(false);

  await user.click(accordionButton);
  expect(onOpen).toHaveBeenCalled();
  checkAccordionPanelVisibility(true);
  expect(screen.getAllByRole('link')).toHaveLength(2);

  await user.click(accordionButton);
  checkAccordionPanelVisibility(false);
  expect(onClose).toHaveBeenCalled();
});

it('should render with an icon', () => {
  const { container } = setupSidebarNavigationAccordionItem({ Icon: IconBranch });
  expect(container).toMatchSnapshot();
});

it("shouldn't have any a11y violation", async () => {
  const { container, user } = setupSidebarNavigationAccordionItem({ Icon: IconBranch });
  await expect(container).toHaveNoA11yViolations();

  await user.click(screen.getByRole('button'));
  await expect(container).toHaveNoA11yViolations();
});

describe('integration with SidebarNavigationItem', () => {
  it('should have active class and rely on the css property passed set form the accordion', () => {
    setupSidebarNavigationAccordionItem({
      children: (
        <>
          <SidebarNavigationItem to="/initial">Sub Item 1</SidebarNavigationItem>
          <SidebarNavigationItem to="/sub-item-2">Sub Item 2</SidebarNavigationItem>
        </>
      ),
    });

    expect(screen.getByRole('link', { name: 'Sub Item 1' })).toHaveStyleRule(
      'display',
      'var(--sidebar-navigation-accordion-children-display, flex)',
    );
    expect(screen.getByRole('link', { name: 'Sub Item 1' })).toHaveClass('active');

    expect(screen.queryByRole('link', { name: 'Sub Item 2' })).toHaveStyleRule(
      'display',
      'var(--sidebar-navigation-accordion-children-display, flex)',
    );
    expect(screen.getByRole('link', { name: 'Sub Item 2' })).not.toHaveClass('active');
  });
});

function checkAccordionPanelVisibility(isOpen: boolean) {
  const region = screen.getByRole('region', { name: 'Accordion Item' });
  expect(region).toHaveAttribute('data-accordion-open', isOpen.toString());
}

function setupSidebarNavigationAccordionItem(
  props: Partial<SidebarNavigationAccordionItemProps> = {},
) {
  return renderWithMemoryRouter(
    <SidebarNavigationAccordionItem label="Accordion Item" {...props}>
      {props.children ?? (
        <>
          <SidebarNavigationItem isActive to="/sub-item-1">
            Sub Item 1
          </SidebarNavigationItem>
          <SidebarNavigationItem to="/sub-item-2">Sub Item 2</SidebarNavigationItem>
        </>
      )}
    </SidebarNavigationAccordionItem>,
  );
}
