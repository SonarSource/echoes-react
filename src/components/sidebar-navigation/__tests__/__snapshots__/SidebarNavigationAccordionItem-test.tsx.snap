// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`should render with an icon 1`] = `
.emotion-0 {
  all: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--echoes-dimension-space-100);
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: hidden;
  box-sizing: border-box;
  height: var(--echoes-dimension-height-800);
  padding: var(--echoes-dimension-space-100);
  border-radius: var(--echoes-border-radius-400);
  background-color: var(--echoes-color-surface-default);
  color: var(--echoes-color-text-default);
  font: var(--echoes-typography-text-default-regular);
  cursor: pointer;
}

.emotion-0:hover {
  background-color: var(--echoes-color-surface-hover);
}

.emotion-0:focus-visible {
  outline: var(--echoes-color-focus-default) solid var(--echoes-focus-border-width-default);
  outline-offset: var(--echoes-focus-border-offset-default);
  border-radius: var(--echoes-border-radius-400);
}

.emotion-0:active {
  background-color: var(--echoes-color-surface-active);
}

.emotion-2 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  font-size: calc(1em + 4px);
  font-style: normal;
  font-weight: normal;
  height: calc(2em - 16px);
  line-height: calc(2em - 16px);
  text-align: center;
  vertical-align: bottom;
  width: calc(2em - 16px);
  font-family: 'Echoes';
}

.emotion-4 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-6 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  font-size: calc(1em + 4px);
  font-style: normal;
  font-weight: normal;
  height: calc(2em - 16px);
  line-height: calc(2em - 16px);
  text-align: center;
  vertical-align: bottom;
  width: calc(2em - 16px);
  font-family: 'Material Symbols Rounded';
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--echoes-dimension-space-50);
  --sidebar-navigation-accordion-children-display: flex;
}

[data-sidebar-collapsed='false'] .emotion-8 {
  margin-left: var(--echoes-dimension-space-300);
  padding-left: var(--echoes-dimension-space-100);
  padding-right: var(--echoes-dimension-space-200);
  border-left: var(--echoes-border-width-default) solid var(--echoes-color-border-weak);
}

.emotion-8[data-accordion-open='false'],
[data-sidebar-collapsed='true'] .emotion-8 {
  --sidebar-navigation-accordion-children-display: none;
}

.emotion-10 {
  all: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--echoes-dimension-space-100);
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: hidden;
  box-sizing: border-box;
  height: var(--echoes-dimension-height-800);
  padding: var(--echoes-dimension-space-100);
  border-radius: var(--echoes-border-radius-400);
  background-color: var(--echoes-color-surface-default);
  color: var(--echoes-color-text-default);
  font: var(--echoes-typography-text-default-regular);
  cursor: pointer;
  display: var(--sidebar-navigation-accordion-children-display, flex);
}

.emotion-10:hover {
  background-color: var(--echoes-color-surface-hover);
}

.emotion-10:focus-visible {
  outline: var(--echoes-color-focus-default) solid var(--echoes-focus-border-width-default);
  outline-offset: var(--echoes-focus-border-offset-default);
  border-radius: var(--echoes-border-radius-400);
}

.emotion-10:active,
.emotion-10.active {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: var(--echoes-color-background-selected-weak-default);
  color: var(--echoes-color-text-accent);
  font: var(--echoes-typography-text-default-semi-bold);
}

.emotion-10:active:hover,
.emotion-10.active:hover {
  background-color: var(--echoes-color-background-selected-weak-hover);
}

<div>
  <button
    aria-controls=":r1:sidebar-accordion-panel"
    aria-expanded="false"
    class="emotion-0 emotion-1"
    id=":r1:sidebar-accordion"
  >
    <span
      aria-hidden="true"
      class="emotion-2 emotion-3"
      css="[object Object]"
    >
      !
    </span>
    <span
      class="emotion-4 emotion-5"
    >
      Accordion Item
    </span>
    <span
      aria-hidden="true"
      class="emotion-6 emotion-7"
      css="[object Object]"
    >
      î—Œ
    </span>
  </button>
  <section
    aria-labelledby=":r1:sidebar-accordion"
    class="emotion-8 emotion-9"
    data-accordion-open="false"
    id=":r1:sidebar-accordion-panel"
  >
    <a
      class="active emotion-10 emotion-11"
      href="/sub-item-1"
    >
      <span
        class="emotion-4 emotion-5"
      >
        Sub Item 1
      </span>
    </a>
    <a
      class="emotion-10 emotion-11"
      href="/sub-item-2"
    >
      <span
        class="emotion-4 emotion-5"
      >
        Sub Item 2
      </span>
    </a>
  </section>
  <pre>
    /initial
  </pre>
  <section
    aria-atomic="false"
    aria-label="Focus toasts messages with alt+T"
    aria-live="polite"
    aria-relevant="additions text"
    tabindex="-1"
  />
</div>
`;
